src/Symfony/Component/HttpFoundation/Tests/RateLimiter/AbstractRequestRateLimiterTest.php

```diff
@@ -33,6 +34,30 @@ public function testConsume(array $rateLimits, ?RateLimit $expected)
         $this->assertSame($expected, $rateLimiter->consume(new Request()));
     }
 
+    public function testConsumeWithoutLimiterAddsSpecialNoLimiter()
+    {
+        $rateLimiter = new MockAbstractRequestRateLimiter([]);
+
+        try {
+            $this->assertSame(\PHP_INT_MAX, $rateLimiter->consume(new Request())->getLimit());
+        } catch (\TypeError $error) {
+            $this->markTestSkipped('This test cannot be run on a version of the RateLimiter component that uses \INF instead of \PHP_INT_MAX in NoLimiter.');
```

Comment:
This would swallow any TypeError. Not a good thing.

Reply:
Right, I added a check on the error message. If I can recall, something like this has already been done elsewhere in the code.